geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java

```diff
@@ -1316,6 +1320,14 @@ protected void txApplyEntryOp(FarSideEntryOp entryOp, List<EntryEventImpl> pendi
       }
     }
 
+    boolean isOpDestroyEvent(InternalRegion internalRegion, FarSideEntryOp entryOp) {
```

Comment:
I think we need a better name for this method. It is too close to what we already know is true: "op.isDestroy".
Also it does not need to test "entryOp.op.isDestroy()" again because we just did it on line 1299.
How about moving lines 1300..1303 and this method into LocalRegion.txApplyDestroy?
We already pass it entryOp.key and it is the "internalRegion".  Also why do you introduce the new variable name "whichPendingCallbacks"? Why not just say something like:
  if (!needsPendingCallbacksForDestroy(key)) {
    pendingCallbacks = new ArrayList<>();
  }

  On LocalRegion you could have this:
  private boolean needsPendingCallbacksForDestroy(Object key) {
    if (isProxy()) {
      return true;
    }
   RegionEntry regionEntry = basicGetEntry(key);
    return regionEntry != null && !regionEntry.isDestroyedOrRemoved();
  }
It would be nice to avoid the getEntry call if it is a proxy.
It is very important to not call getValue here just to check if it is removed. For things like offheap and compression getValue does a bunch of work. isDestroyedOrRemoved uses getValueAsToken which is all we care about here since we are just trying to confirm it is not one of the removed tokens.